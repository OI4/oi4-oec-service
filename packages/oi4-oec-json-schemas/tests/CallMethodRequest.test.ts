import { matchersWithOptions } from 'jest-json-schema'
import schema from '../src/schemas/CallMethodRequest.schema.json'
import BaseDataType from '../src/schemas/BaseDataType.schema.json'

expect.extend(
  matchersWithOptions({
    // Loading in a schema which is comprised only of definitions,
    // which means specific test schemas need to be created.
    // This is good for testing specific conditions for definition schemas.
    schemas: [BaseDataType],
    verbose: true,
  })
)

describe('CallMethodRequest schema', () => {
  it('validate schema', () => {
    expect(schema).toBeValidSchema()
  })

  it('test valid values', () => {
    expect({ methodId: 'a', inputArguments: [''] }).toMatchSchema(schema)
    expect({ methodId: '1', inputArguments: [1] }).toMatchSchema(schema)
    expect({ methodId: 'a', inputArguments: [true] }).toMatchSchema(schema)
    expect({ methodId: 'b', inputArguments: [{}] }).toMatchSchema(schema)
    expect({ methodId: 'c', inputArguments: [{ a: 'b', b: 1 }] }).toMatchSchema(
      schema
    )
    expect({ methodId: 'c', inputArguments: [[]] }).toMatchSchema(schema)
    expect({ methodId: 'c', inputArguments: [[1]] }).toMatchSchema(schema)
    expect({ methodId: 'c', inputArguments: [['']] }).toMatchSchema(schema)
    expect({ methodId: 'c', inputArguments: [[{}]] }).toMatchSchema(schema)
    expect({
      methodId: 'c',
      inputArguments: [[{ a: 'b', b: 1 }]],
    }).toMatchSchema(schema)
  })

  it('test invalid values', () => {
    const invalidValues = [
      {},
      { methodId: 'a' },
      { inputArguments: '' },
      { methodId: '', inputArguments: [] },
      { methodId: 1, namespaceIndex: 1 },
    ]

    invalidValues.forEach((value, index) => {
      expect(value).not.toMatchSchema(schema)
      expect(() => {
        expect(value).toMatchSchema(schema)
      }).toThrowErrorMatchingSnapshot(index.toString())
    })
  })
})
