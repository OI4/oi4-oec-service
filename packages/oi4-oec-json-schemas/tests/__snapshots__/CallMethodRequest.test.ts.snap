// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CallMethodRequest schema test invalid values: 0 1`] = `
"[2mexpect([22m[31mreceived[39m[2m).toMatchSchema([22m[32mschema[39m[2m)[22m

received
[31m  must have required property 'inputArguments'[39m
[31m    Path:     CallMethodRequest.schema.json#/required[39m
[31m[39m[31m  must have required property 'methodId'[39m
[31m    Path:     CallMethodRequest.schema.json#/required[39m
[31m[39m"
`;

exports[`CallMethodRequest schema test invalid values: 1 1`] = `
"[2mexpect([22m[31mreceived[39m[2m).toMatchSchema([22m[32mschema[39m[2m)[22m

received
[31m  must have required property 'inputArguments'[39m
[31m    Path:     CallMethodRequest.schema.json#/required[39m
[31m[39m"
`;

exports[`CallMethodRequest schema test invalid values: 2 1`] = `
"[2mexpect([22m[31mreceived[39m[2m).toMatchSchema([22m[32mschema[39m[2m)[22m

received
[31m  must have required property 'methodId'[39m
[31m    Path:     CallMethodRequest.schema.json#/required[39m
[31m[39m[31m  /inputArguments must be array[39m
[31m    Received: [33m<empty string>[39m[31m[39m
[31m    Path:     CallMethodRequest.schema.json#/properties/inputArguments/type[39m
[31m[39m"
`;

exports[`CallMethodRequest schema test invalid values: 3 1`] = `
"[2mexpect([22m[31mreceived[39m[2m).toMatchSchema([22m[32mschema[39m[2m)[22m

received
[31m  /methodId must NOT have fewer than 1 characters[39m
[31m    Received: [33m<empty string>[39m[31m[39m
[31m    Path:     CallMethodRequest.schema.json#/properties/methodId/minLength[39m
[31m[39m"
`;

exports[`CallMethodRequest schema test invalid values: 4 1`] = `
"[2mexpect([22m[31mreceived[39m[2m).toMatchSchema([22m[32mschema[39m[2m)[22m

received
[31m  must have required property 'inputArguments'[39m
[31m    Path:     CallMethodRequest.schema.json#/required[39m
[31m[39m[31m  must NOT have additional properties, but found 'namespaceIndex'[39m
[31m    Expected: [33m<boolean>[39m[31m false[39m
[31m    Path:     CallMethodRequest.schema.json#/additionalProperties[39m
[31m[39m[31m  /methodId must be string[39m
[31m    Received: [33m<number>[39m[31m 1[39m
[31m    Path:     CallMethodRequest.schema.json#/properties/methodId/type[39m
[31m[39m"
`;
